src/scan_nm2.cc

```diff
@@ -0,0 +1,353 @@
+            if (std::filesystem::exists(check) && access(check.c_str(), X_OK) == 0) {
+                return (path / name).string();
+            }
+        }
+
+        return fty::unexpected("Executable {} was not found", name);
+    }
+
+private:
+    std::string                           m_address;
+    std::vector<secw::UserAndPasswordPtr> m_creds;
+    fty::nut::KeyValues*                  m_mapping = nullptr;
+    fty::nut::KeyValues*                  m_sensorMapping;
+};
+
+// =========================================================================================================================================
+
+struct AutoRemove
+{
+    template <typename FuncT>
+    AutoRemove(FuncT&& func)
+        : m_func(func)
+    {
+    }
+    ~AutoRemove()
+    {
+        m_func();
+    }
+    std::function<void()> m_func;
+};
+
+// =========================================================================================================================================
+
+static bool askActorTerm(zsock_t* pipe)
+{
+    zmsg_t* msg_stop = zmsg_recv_nowait(pipe);
+    if (msg_stop) {
+        char* cmd = zmsg_popstr(msg_stop);
+        if (cmd && streq(cmd, "$TERM")) {
+            zstr_free(&cmd);
+            zmsg_destroy(&msg_stop);
+            return true;
+        }
+        zstr_free(&cmd);
+        zmsg_destroy(&msg_stop);
+    }
+    return false;
+}
+
+// =========================================================================================================================================
+
+void scan_nm2_actor(zsock_t* pipe, void* args)
+{
+    zsock_signal(pipe, 0);
+    AutoRemove cleanup([&]() {
+        if (args) {
+            zlist_t* argv = static_cast<zlist_t*>(args);
+            zlist_destroy(&argv);
+        }
+        zmsg_t* reply = zmsg_new();
+        zmsg_pushstr(reply, REQ_DONE);
+        zmsg_send(&reply, pipe);
+    });
+
+    if (!args) {
+        logError("{} : actor created without parameters", __FUNCTION__);
+        return;
+    }
+
+    zlist_t* argv = static_cast<zlist_t*>(args);
+    if (!argv || zlist_size(argv) != 5) {
+        log_error("{} : actor created without config or devices list", __FUNCTION__);
+        return;
+    }
+
+    CIDRList*              listAddr      = static_cast<CIDRList*>(zlist_first(argv));
+    discovered_devices_t*  devices       = static_cast<discovered_devices_t*>(zlist_next(argv));
+    std::set<std::string>* documentIds   = static_cast<std::set<std::string>*>(zlist_next(argv));
+    fty::nut::KeyValues*   mapping       = static_cast<fty::nut::KeyValues*>(zlist_next(argv));
+    fty::nut::KeyValues*   sensorMapping = static_cast<fty::nut::KeyValues*>(zlist_next(argv));
+
+    if (!listAddr || !devices || !documentIds || !mapping || !sensorMapping) {
+        logError("{} : actor created without config or devices list", __FUNCTION__);
+        return;
+    }
+
+    fty::SocketSyncClient secwSyncClient("/run/fty-security-wallet/secw.socket");
```

Comment:
that should also be a conf file param

Reply:
About a config, you are right, but will this project finish next release? This conf file is weird to patch and provide changes to images (looks like those configs somewhere in different places).